Wrappers/Python/cil/utilities/errors.py

```diff
@@ -0,0 +1,6 @@
+
+class InPlaceError(Exception):
+    "Raised when the object should not be used in-place"
+    def __init__(self,  message="Object should not be used in-place"):
```

Comment:
Is there a way to make it clear which object the error has occurred in?

Reply:
Good question - I am not sure. 

When the error is raised it does traceback where the error was raised e.g. 
![image](https://github.com/TomographicImaging/CIL/assets/43645617/1f2289c3-5426-44d5-b464-e055ba84c423)
The user can see what call raised the error.

Reply:
I think you can redefine the error message each time you raise the error, something along these lines:
```
raise InPlaceError('TotalVariation.proximal cannot be used in place')
```

Reply:
Do you know how @paskino or I can search for it?

Reply:
I think I have done this